---
title: Descriptive statistics of read counts
---

```{r libraries}
#| results: hide
.packages <- c("knitr", "tidyverse", "kableExtra")
.installed <- .packages %in% installed.packages()
if (length(.packages[!.installed])>0) install.packages(.packages[!.installed])
lapply(.packages, library, character.only=T)
# edgeR installation
if (!require("edgeR", quietly = TRUE))
    BiocManager::install(c("edgeR"))
library(edgeR)
```

```{r base-colors}
color1 <- "#00BFC4"
color2 <- "#F8766D"
color3 <- "#7CAE00"
color4 <- "#C77CFF"
```

```{r data-loading}
# reads counts summary
source("preprocesing.R")
```

```{r normalization}
# TMM Normalization
# Single-cell RNA-seq requires a specialized normalization method
reads_filtered <- calcNormFactors(reads_filtered)
#Normalization factors
#reads_filtered$samples
```

## Quasi-likelihood F-test

```{r}
design <- model.matrix(~df_groups$Group)
reads_filtered <- estimateDisp(reads_filtered, design)
fit <- glmQLFit(reads_filtered, design)
qlf <- glmQLFTest(fit, coef=2)
result <- as.data.frame(topTags(qlf), n=nrow(matrix_reads))
result %>% kable(caption = paste0("Genes with a significant differential expression")) %>%
  kable_styling(bootstrap_options = "hover", full_width = F)
```

```{r}
# Design matrix not provided.
# reads_filtered <- estimateDisp(reads_filtered)
# et <- exactTest(reads_filtered)
# result <- as.data.frame(topTags(et), n=nrow(matrix_reads))
# result
```

## Means difference plots

```{r means-differences-plot}
for (i in 1:21) {
    plotMD(reads_filtered, column = i)
    abline(h=0, col="red", lty = 2, lwd = 2) 
}
```